samples/performance/constant_data/constant_data_tutorial.md

```diff
@@ -0,0 +1,403 @@
+
+For example, this `vkCmdDrawIndexed` is taken from the [Vulkan spec](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdDrawIndexed.html):
+
+```
+void vkCmdDrawIndexed(
+    VkCommandBuffer                             commandBuffer,
+    uint32_t                                    indexCount,
+    uint32_t                                    instanceCount,
+    uint32_t                                    firstIndex,
+    int32_t                                     vertexOffset,
+    uint32_t                                    firstInstance);
+```
+
+Here is some pseudo-code to show how the `vkCmdDrawIndexed` function is used, and also to describe how a generic scene render function will look:
+
+```
+uint32_t instance_index = 0;
+
+vkCmdBindDescriptorSet(command_buffer, buffer_array_descriptor_set);
+
+for(auto &mesh : meshes)
+{
+    vkCmdBindVertexBuffer(command_buffer, mesh.vertex_buffer);
+    vkCmdBindIndexBuffer(command_buffer, mesh.index_buffer);
+
+    // This line does our drawing
+    vkCmdDrawIndexed(command_buffer, mesh.index_buffer.size(), 1, 0, 0, instance_index++);
+}
+```
+
+In the code snippet above we can see that we bind our descriptor set once, and for each mesh bind its vertex and index buffers and then execute a draw call with an incrementing value for `uint32_t firstInstance`. This `uint32_t` will be substituted in wherever `gl_InstanceIndex` exists in the shader code, which will pull out the required model matrix to position the mesh inside our world.
+
+
+### **Performance**
+
+While this could be a fast method for some devices, on Mali it is not a recommend practice as it disables a compiler optimisation technique known as **pilot shaders**.
+
+Pilot shaders are a technique that allows us to determine what calculations can be "piloted" into your GPU's register so that when the data needs to be read it doesn't take a full read cycle from the GPU RAM.
+
+To show this here is a Streamline capture of a Mali G76, showing the read cycles for using a single descriptor set per mesh against the pre allocated buffer array:
+
+![loadcycles](images/loadcycles.png)
+
+A few different stats are affected in the Mali GPU by using this, but the main thing is the **full read** in the **Mali Core Load/Store Cycles**.
+
+## **Further reading**
+
+* The [Vulkan 1.2 spec](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html)
+* "Writing an efficient Vulkan renderer" by Arseny Kapoulkine https://zeux.io/2020/02/27/writing-an-efficient-vulkan-renderer/
+* Alexander Overvoorde's [Vulkan Tutorial on Descriptors](https://vulkan-tutorial.com/Uniform_buffers/Descriptor_layout_and_buffer) guide
+* Vulkan Fast Paths http://32ipi028l5q82yhj72224m8j-wpengine.netdna-ssl.com/wp-content/uploads/2016/03/VulkanFastPaths.pdf
+* Performance root signature of descriptor sets https://gpuopen.com/performance-root-signature-descriptor-sets/
```

Comment:
It seems that they have reworked the whole gpuopen.com page, and the link no longer works. Haven't been able to find that page via the search, so maybe remove that link.

Reply:
Will remove